dev-tools/prepare_release_candidate.py

```diff
@@ -0,0 +1,271 @@
+                      help='Only runs a maven install to skip the remove deployment step')
+  parser.add_argument('--gpg-key', '-k', dest='gpg_key', default="D88E42B4",
+                      help='Allows you to specify a different gpg_key to be used instead of the default release key')
+  parser.set_defaults(deploy=False)
+  parser.set_defaults(skip_doc_check=False)
+  parser.set_defaults(push=False)
+  parser.set_defaults(install_only=False)
+  args = parser.parse_args()
+  install_and_deploy = args.deploy
+  skip_doc_check = args.skip_doc_check
+  push = args.push
+  gpg_key = args.gpg_key
+  install_only = args.install_only
+
+  ensure_checkout_is_clean()
+  release_version = find_release_version()
+
+  print('*** Preparing release version: [%s]' % release_version)
+
+  if not skip_doc_check:
+    print('*** Check for pending documentation changes')
+    pending_files = update_reference_docs(release_version)
+    if pending_files:
+      raise RuntimeError('pending coming[%s] documentation changes found in %s' % (release_version, pending_files))
+
+
+  run('cd dev-tools && mvn versions:set -DnewVersion=%s -DgenerateBackupPoms=false' % (release_version))
+  run('cd rest-api-spec && mvn versions:set -DnewVersion=%s -DgenerateBackupPoms=false' % (release_version))
+  run('mvn versions:set -DnewVersion=%s -DgenerateBackupPoms=false' % (release_version))
+
+  remove_version_snapshot(VERSION_FILE, release_version)
+
+  print('*** Done removing snapshot version. DO NOT COMMIT THIS, WHEN CREATING A RELEASE CANDIDATE.')
+
+  shortHash = subprocess.check_output('git log --pretty=format:"%h" -n 1', shell=True).decode('utf-8')
+  localRepo = '/tmp/elasticsearch-%s-%s' % (release_version, shortHash)
+  localRepoElasticsearch = localRepo + '/org/elasticsearch'
+  if os.path.exists(localRepoElasticsearch):
+    print('clean local repository %s' % localRepoElasticsearch)
+    shutil.rmtree(localRepoElasticsearch)
+
+  if install_only:
+    mvn_targets = 'install'
+  else:
+    mvn_targets = 'install deploy'
+  install_command = 'mvn clean %s -Prelease -Dskip.integ.tests=true -Dgpg.keyname="%s" -Dpackaging.rpm.rpmbuild=/usr/bin/rpmbuild -Drpm.sign=true -Dmaven.repo.local=%s -Dno.commit.pattern="\\bno(n|)commit\\b" -Dforbidden.test.signatures=""' % (mvn_targets, gpg_key, localRepo)
+  clean_repo_command = 'find %s -name _remote.repositories -exec rm {} \;' % (localRepoElasticsearch)
+  rename_metadata_files_command = 'for i in $(find %s -name "maven-metadata-local.xml*") ; do mv "$i" "${i/-local/}" ; done' % (localRepoElasticsearch)
+  s3_sync_command = 's3cmd sync %s s3://download.elasticsearch.org/elasticsearch/staging/elasticsearch-%s-%s/org/' % (localRepoElasticsearch, release_version, shortHash)
+  s3_bucket_sync_to = 'download.elasticsearch.org/elasticsearch/staging/elasticsearch-%s-%s/repos' % (release_version, shortHash)
+  build_repo_command = 'dev-tools/build_repositories.sh %s' % (release_version)
```

Comment:
I think we have a problem here. The version that is supposed to be supplied as a parameter should only consist of major.minor version like `2.0` (so that all 2.x version go into the same repository) - this one is `2.0.0-beta1` though.